### DISCUSSION TITLE
Get instance of queue from worker process

### DISCUSSION THREAD

**Discussion by mattweberio** [taskforcesh/bullmq#1722] [Q&A]
Hello, 

We must drain the queue if a job fails because every job depends on the previous job.

I've tried to get a reference to the Queue from inside the worker to execute the drain method.

e.g. 
`const queue = new Queue(sessionId)`

But it appears to create a new instance and does not connect to the same queue.  For example, when I call drain(), it doesn't work, and getActiveCount() always returns 0.

What is the correct way to clear the queue of all jobs if one job fails?

Thank you!

**Comment by manast** [comment]
Are you connecting to the same Redis host?

**Reply by mattweberio** [reply]
Hi @manast - Thank you for the reply.  Yes, it's the same Redis host because the event is from inside the worker process.  The worker is already processing the job queue.  From inside the process, I want to stop processing and clear the remaining jobs.  By your reply, it sounds like perhaps the "new Queue" is meant to return the existing instance if it has the name?

**Reply by manast** [reply]
new Queue does not imply a new queue in Redis, it means a new Queue instance, otherwise it would not be possible to add jobs and process them from many processes.
